### PR TITLE
Publish downtime messages sent by bot

### PR DESCRIPTION
**Pull request by ellieisjelly**
Noticed this earlier, tested working on announcement channels

### CODE REVIEW

**Review by NoComment1105 — CHANGES_REQUESTED** [review]
You'll need to do some extra verification to find out if the channels are actually announcement channels, since I've left the ability to use any channel, theres a chance it will just raise an exception

**Comment by tempest15** [comment]
It would be nice if the publishing was a config option when setting the  bot to be watched, but you would still need to check later as well

**Review by ellieisjelly — COMMENTED** [review]
Fixed

File: src/main/kotlin/org/hyacinthbots/saffronstatus/database/migrations/v1.kt [hunk 1]
```diff
@@ -6,9 +6,20 @@
...
 package org.hyacinthbots.saffronstatus.database.migrations
 
+import org.hyacinthbots.saffronstatus.database.entities.WatchedBotData
 import org.litote.kmongo.coroutine.CoroutineDatabase
+import org.litote.kmongo.exists
+import org.litote.kmongo.setValue
 
 @Suppress("UnusedPrivateMember")
 suspend fun v1(db: CoroutineDatabase) {
 	// TODO Fill with migration and set true in main function
```
Comment (NoComment1105) [inline src/main/kotlin/org/hyacinthbots/saffronstatus/database/migrations/v1.kt]: 🦀  this

File: src/main/kotlin/org/hyacinthbots/saffronstatus/database/migrations/v1.kt [hunk 2]
```diff
@@ -6,9 +6,20 @@
...
+	with(db.getCollection<WatchedBotData>()) {
+		updateMany(
+			WatchedBotData::publishMessage exists false,
+			    setValue(
+			        WatchedBotData::publishMessage, false
+			)
```
Comment (NoComment1105) [inline src/main/kotlin/org/hyacinthbots/saffronstatus/database/migrations/v1.kt]: Some strange indentation afoot here. Try ctrl + alt + L or the below
```suggestion
			WatchedBotData::publishMessage exists false,
			setValue(WatchedBotData::publishMessage, false)
```

**Review by NoComment1105 — CHANGES_REQUESTED** [review]
Nearly there, very good

File: src/main/kotlin/org/hyacinthbots/saffronstatus/extensions/WatchedBot.kt [hunk 3]
```diff
@@ -190,7 +192,10 @@ class WatchedBot : Extension() {
 			name = "notification-channel"
 			description = "The channel to send the notifications too when the bot goes offline"
 		}
-
+		val publishMessage by boolean {
+			name = "publish-message"
+			description = "Whether to publish the downtime messages for this bot"
```
Comment (tempest15) [inline src/main/kotlin/org/hyacinthbots/saffronstatus/extensions/WatchedBot.kt]:
```suggestion
			description = "If downtime messages sent in announcement channels should be published"
```

**Review by tempest15 — APPROVED** [review]
LGTM, we'll wait for NC to merge

**Review by NoComment1105 — APPROVED** [review]

**Comment by NoComment1105** [comment]
Apparently never merged this lmao

**Pull request merged**